### ISSUE TITLE
Asp.Net Core Error handling

### ISSUE THREAD

**Issue by HeyJoel** [cofoundry-cms/cofoundry#113]
This hasn't been looked at in much detail yet but I know it's changed so much in asp.net core that we need to look at the whole thing again. 

Some of the old code is in `ExceptionLogAttribute`, also check out `Error.cshtml` in various places and think about how this used to change in dev mode to show the exception. 

Some resources:

- https://dusted.codes/error-handling-in-aspnet-core
- https://andrewlock.net/retrieving-the-path-that-generated-an-error-with-the-statuscodepages-middleware/

Additionally this would be a good time to think about whether `IErrorLoggingService` should be async.

**Comment by HeyJoel** [comment]
- `IErrorLoggingService` has been removed in preference to using `ILogger`
- `IErrorLoggingService` is still available in the Cofoundry.Plugins.ErrorLogging package but is relegated to simply logging exceptions at the end of the asp.net pipeline.
- Developer exceptions will now be handled by the asp.net framework (`UseDeveloperExceptionPage()`)

What still needs working out is whether we bundle in an error page in with CMS in the way we do currently with error.cshtml

**Comment by HeyJoel** [comment]
Custom error pages now work as expected, documentation has been added to the 0.2 docs should will appear [here](https://github.com/cofoundry-cms/cofoundry/wiki/Custom-Error-Pages) when merged in on release.

We needed to take control of the error handling middleware to make this work so this is now added automatically in a startup task. This means that any references to error handling middleware such as  `UseDeveloperExceptionPage()` should be removed from applications that use Cofoundry.
